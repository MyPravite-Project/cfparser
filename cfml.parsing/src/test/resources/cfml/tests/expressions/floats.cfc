x = 1.23;
y = -33.0;
z = 2.;
a = 121E10;
b = .12;