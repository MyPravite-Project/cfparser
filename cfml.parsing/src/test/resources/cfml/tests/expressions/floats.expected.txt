/*===TOKENS===*/
IDENTIFIER               <x>
'='                      <=>
INTEGER_LITERAL          <1>
'.'                      <.>
INTEGER_LITERAL          <23>
';'                      <;>
IDENTIFIER               <y>
'='                      <=>
'-'                      <->
INTEGER_LITERAL          <33>
'.'                      <.>
INTEGER_LITERAL          <0>
';'                      <;>
IDENTIFIER               <z>
'='                      <=>
INTEGER_LITERAL          <2>
'.'                      <.>
';'                      <;>
IDENTIFIER               <a>
'='                      <=>
FLOATING_POINT_LITERAL   <121E10>
';'                      <;>
IDENTIFIER               <b>
'='                      <=>
'.'                      <.>
INTEGER_LITERAL          <12>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier x))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression (literalExpression (floatingPointExpression 1 . 23)))
              )
            )
          )
        )
      )
      ;
    )
  )
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier y))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                -
                (primaryExpression (literalExpression (floatingPointExpression 33 . 0)))
              )
            )
          )
        )
      )
      ;
    )
  )
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier z))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression (literalExpression (floatingPointExpression 2 .)))
              )
            )
          )
        )
      )
      ;
    )
  )
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier a))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression (literalExpression (floatingPointExpression 121E10)))
              )
            )
          )
        )
      )
      ;
    )
  )
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier b))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression (literalExpression (floatingPointExpression . 12)))
              )
            )
          )
        )
      )
      ;
    )
  )
)
/*======*/
/*===DECOMPILE===*/
{
x = 1.23;
y = -33.0;
z = 2.;
a = 121E10;
b = .12;

}
/*======*/